modules/Bio/EnsEMBL/DBSQL/AltAlleleGroupAdaptor.pm

```diff
@@ -231,20 +231,56 @@ sub fetch_by_dbID {
 sub fetch_by_gene_id {
     my ($self, $gene_id) = @_;
 
+    my $aag_list = $self->fetch_all_by_gene_id($gene_id);
+
+    # return first group from list
+    my $aag = @$aag_list[0];
+    # check that a result exists
+    if ($aag) {
+        my $group_id = $aag->dbID;
+        return $self->fetch_by_dbID($group_id);
+    }
+    return;
+}
+
+=head2 fetch_all_by_gene_id
+
+  Arg[1]      : Integer Gene ID of the member to query by
+  Description : Creates and returns an array of one or more AltAlleleGroups, 
+                which each contain the specified gene member                
+  Returntype  : Array of Bio::EnsEMBL::AltAlleleGroup objects
+
+=cut
+
+sub fetch_all_by_gene_id {
+    my ($self, $gene_id) = @_;
+
     my $gene_id_sql = q(
         SELECT alt_allele_group_id FROM alt_allele
-        WHERE gene_id = ?
+        WHERE gene_id = ? ORDER BY alt_allele_group_id
     );
     my $sth = $self->prepare($gene_id_sql);
     $sth->bind_param(1,$gene_id, SQL_INTEGER);
     
+    my @group_ids;
     my $group_id;
     $sth->execute();
     $sth->bind_col(1,\$group_id);
-    $sth->fetch;
+    while ( $sth->fetch ) {
+        # print "group id: $group_id\n";
+        push( @group_ids, $group_id );
+    }
     $sth->finish;
     if (!$@ && $group_id) {
-        return $self->fetch_by_dbID($group_id);
+        my @aag;
+        foreach my $group (@group_ids) {
+            # print "About to fetch aag object using group id $group\n";
```

Comment:
Please remove this and the other debug comments.

Reply:
Removed, thanks